### PR TITLE
Remove leftover `clean_logger` call in tests

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #10896 

The `_clean_logger` method was removed in #10896 and the test here was still using it, not failing of course because who would expect a test to fail due to AttributeError, right?  

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @justusschock @awaelchli @akihironitta @borda

### CODE REVIEW

**Review by rohitgr7 — APPROVED** [review]

File: tests/plugins/test_tpu_spawn.py [hunk 1]
```diff
@@ -114,7 +114,6 @@ def test_tensorboard_ddp_spawn_cleanup(use_list, tmpdir):
     assert tensorboard_logger._experiment is not None
```
Comment (carmocca) [inline tests/plugins/test_tpu_spawn.py:114]: I think we should remove the test entirely. Its only purpose was to test `_clean_logger`

File: tests/plugins/test_tpu_spawn.py [hunk 2]
```diff
@@ -114,7 +114,6 @@ def test_tensorboard_ddp_spawn_cleanup(use_list, tmpdir):
...
     logger = [tensorboard_logger] if use_list else tensorboard_logger
     trainer = Trainer(strategy="ddp_spawn", accelerator="tpu", devices="auto", logger=logger)
-    trainer.training_type_plugin._clean_logger(trainer)
```
Comment (carmocca) [inline tests/plugins/test_tpu_spawn.py:117]: Wait do you know why this did not fail? Is it because it doesn't run at all?

**Review by carmocca — APPROVED** [review]

File: tests/plugins/test_tpu_spawn.py (hunk 2 above)
Comment (awaelchli) [reply tests/plugins/test_tpu_spawn.py:117]: there is a file test_tpu_cases.jsonnet and it does not include the file these tests reside in. 
So yeah, RIP.

Comment (carmocca) [reply tests/plugins/test_tpu_spawn.py:117]: I guess we should replicate the test search and filtering we do for `standalone=True` but for `tpu=True`

**Review by ananthsub — APPROVED** [review]

File: tests/plugins/test_tpu_spawn.py (hunk 2 above)
Comment (awaelchli) [reply tests/plugins/test_tpu_spawn.py:117]: doing that in #11098 👍

**Review by carmocca — APPROVED** [review]
